### PR TITLE
[ci skip] Update routing guide for Direct & resolved routes

### PR DESCRIPTION
**Pull request by yhirano55**
### Summary

* Added the direct method to routing guide.

### CODE REVIEW

**Comment by kamipo** [comment]
Can you also add `resolve` since 5.1 Release Notes refers to both?

**Comment by yhirano55** [comment]
@kamipo Thanks for the comment. I wanted to add `resolve`, but I didn't know whether it should be added where section in routing guide. So in this request, I hope to resolve only the `direct` method.

**Comment by kamipo** [comment]
Both are implemented on `CustomUrlHelper` and define custom url mappings/helpers.
So just place it in the same section I think.

**Comment by yhirano55** [comment]
@kamipo I see. I'll try it. Thanks so much.

File: guides/source/routing.md [hunk 1]
```diff
@@ -852,6 +852,49 @@ You can specify unicode character routes directly. For example:
...
+```
+
+``` erb
+<%= form_for @basket do |form| %>
+  <!-- basket form -->
+<% end %>
+```
+
+This will generate the singular URL `/basket` instead of the usual `/baskets/:id`.
+
```
Comment (yhirano55) [inline guides/source/routing.md]: @kamipo I've added the `resolve` method section. Please code-review again.

**Pull request merged**